tests/xybeamstop.py

```diff
@@ -0,0 +1,164 @@
+import unittest
+import math
+import time
+
+from utils.channel_access import ChannelAccess
+from utils.ioc_launcher import IOCRegister
+
+# Internal Address of device (must be 2 characters)
+ADDRESS = "01"
+
+# MACROS to use for the IOC
+MACROS = {"ADDR": ADDRESS}
+
+PREFIX = "MOT"
+MTR1 = "MOT:MTR0101"
+MTR2 = "MOT:MTR0102"
+
+
+class XybeamstopTests(unittest.TestCase):
+    """
+    Tests for the Larmor X-Y Beamstop
+    """
+    def setUp(self):
+        self._ioc = IOCRegister.get_running("xyBeamstop")
+        self.ca = ChannelAccess()
+        self.ca.wait_for("MOT:ARM:X")
+        self._set_pv_value("ARM:STORE", 0)
+        self._set_x(0.1)
+        self._set_y(0.1)
+
+    def _set_pv_value(self, pv_name, value):
+        self.ca.set_pv_value("{0}:{1}".format(PREFIX, pv_name), value)
+
+    def _set_x(self, value):
+        self._set_pv_value("ARM:X", value)
+
+    def _set_y(self, value):
+        self._set_pv_value("ARM:Y", value)
+
+    def test_GIVEN_x_y_WHEN_read_x_y_THEN_x_y_is_as_expected(self):
+        x = 1.0
+        y = 2.0
+        self._set_x(x)
+        time.sleep(2)
+        self._set_y(y)
+        time.sleep(4)
+
+        self.ca.assert_that_pv_is_close("MOT:ARM:X", x, 1e-2)
+        self.ca.assert_that_pv_is_close("MOT:ARM:Y", y, 1e-2)
+        self.ca.assert_that_pv_is_close("MOT:ARM:X.RBV", x, 1e-1)
+        self.ca.assert_that_pv_is_close("MOT:ARM:Y.RBV", y, 1e-1)
+
+    def test_GIVEN_store_command_WHEN_read_x_y_is_as_expected(self):
+        self._set_pv_value("ARM:STORE", 1)
+        time.sleep(1)
+
+        self.ca.assert_that_pv_is("MOT:ARM:STORE", "STORED")
+        self.ca.assert_that_pv_is_close("MOT:ARM:X", -7.071)
+        self.ca.assert_that_pv_is_close("MOT:ARM:Y", 2.929)
```

Comment:
These positions `-7.071` and `2.929` are used throughout:

1. Where have they come from? It's not clear from the test file why this should be the expected position (a good example of why `as_expected` should never be used in test naming.
1. As the same value is used repeatedly throughout the set of tests, it should be refactored into a class variable, rather than appearing as a magic number

**Please remove all subsequent magic numbers too**